src/components/Snackbar/Snackbar.style.ts

```diff
@@ -0,0 +1,130 @@
+import styled, { css } from 'styled-components';
+import { DefaultTheme } from 'styled-components/dist/types';
+import { match } from 'ts-pattern';
+
+import { SnackbarHeightType, SnackbarPosition, SnackbarProps, SnackbarType } from './Snackbar.type';
+
+interface StyledSnackbarProps {
+  $type: 'info' | 'error';
+  $width: SnackbarProps['width'];
+  $margin?: SnackbarProps['margin'];
+  $isClosing?: boolean;
+  $position: SnackbarPosition;
+  $heightType?: SnackbarHeightType;
+}
+
+const getBackgroundStyle = (arg: { $type: SnackbarType; theme: DefaultTheme }) => {
+  return match(arg)
+    .with({ $type: 'error' }, ({ theme }) => theme.semantic.color.snackbarError)
+    .otherwise(({ theme }) => theme.semantic.color.snackbarInfo);
+};
+
+const getFontColorStyle = (arg: { $type: SnackbarType; theme: DefaultTheme }) => {
+  return match(arg)
+    .with({ $type: 'error' }, ({ theme }) => theme.semantic.color.textStatusNegative)
+    .otherwise(() => arg.theme.semantic.color.textBasicWhite);
+};
+
+const getPositionStyle = (position: SnackbarPosition, margin?: string) => {
+  return match(position)
+    .with(
+      'left',
+      () => css`
+        left: 0;
+        align-items: start;
+        margin-left: ${margin || '16px'};
+      `
+    )
+    .with(
+      'right',
+      () => css`
+        right: 0;
+        align-items: end;
+        margin-right: ${margin || '16px'};
+      `
+    )
+    .otherwise(
+      () => css`
+        left: 0;
+        right: 0;
+        margin: 0 auto;
+        align-items: center;
+      `
+    );
+};
+
+export const StyledSnackbarContainer = styled.div<Omit<StyledSnackbarProps, '$type' | 'visible'>>`
+  position: fixed;
+  bottom: 0;
+  width: ${({ $width }) => ($width === 'full-width' ? '100%' : $width)};
+  height: fit-content;
+  display: flex;
+  flex-direction: column-reverse;
+  margin: 0 auto;
+  align-items: center;
+  ${({ $width, $position, $margin }) =>
+    $width !== 'full-width' && getPositionStyle($position, $margin)}
+`;
+
+export const StyledSnackbar = styled.div.withConfig({
+  shouldForwardProp: (prop) => prop !== 'isClosing',
+})<StyledSnackbarProps>`
+  position: relative;
+  padding: 24px;
+  margin-bottom: 16px;
+  width: ${({ $width }) => ($width === 'full-width' ? 'calc(100% - 32px)' : $width)};
+  height: ${({ $heightType }) => ($heightType === 2 ? '72px' : '52px')};
+  border-radius: ${({ theme }) => theme.semantic.radius.m}px;
+
+  ${({ theme }) => theme.typo.B3_Rg_14}
```

Comment:
info일 때 typo ➡️ PC/Body/B3_Rg_14
**error일 때 typo ➡️ PC/Body/B3_Sb_14**

피그마에 비해 글자가 날씬하길래 보니까 타이포 스타일이 다르더라고요!

![image](https://github.com/user-attachments/assets/fe4962c2-2844-47fd-8d35-952c5e85371b)

```suggestion
  ${({ $type, theme }) => ($type === 'info' ? theme.typo.B3_Rg_14 : theme.typo.B3_Sb_14)}
```

Reply:
ㅋㅋㅋㅋㅋㅋㅋㅋㅋ아 대박 진정한 꼼꼼이는 보리 당신 꼼보...👍🏻

커밋 suggestion 할게영